examples/fhir/MN NBS FHIR Order Message.json

```
@@ -0,0 +1,765 @@
+      "resource": {
+        "resourceType": "Observation",
+        "id": "aba644a3-3666-439e-b207-0693b4fce32e",
+        "status": "final",
+        "category": [
+          {
+            "coding": [
+              {
+                "system": "http://terminology.hl7.org/CodeSystem/observation-category",
+                "code": "vital-signs",
+                "display": "Vital Signs"
+              }
+            ],
+            "text": "Vital Signs"
+          }
+        ],
+        "code": {
+          "coding": [
+            {
+              "system": "http://loinc.org",
+              "code": "57714-8",
+              "display": "OBSTETRIC ESTIMATION OF GESTATIONAL AGE"
+            }
+          ],
+          "text": "OBSTETRIC ESTIMATION OF GESTATIONAL AGE"
+        },
+        "subject": {
+          "reference": "Patient/09a7a7cc-7456-4f09-bdf6-ebc730902fd2",
+          "type": "Patient"
+        },
+        "effectiveDateTime": "2023-05-06T05:00:00-05:00",
+        "valueQuantity": {
+          "value": 32,
+          "unit": "wk",
+          "system": "http://unitsofmeasure.org",
+          "code": "wk"
+        }
+      }
+    },
+    {
+      "fullUrl": "Observation/2bb5a612-68c1-499a-886e-862e85833264",
+      "resource": {
+        "resourceType": "Observation",
+        "id": "2bb5a612-68c1-499a-886e-862e85833264",
+        "status": "final",
+        "code": {
+          "coding": [
+            {
+              "system": "http://loinc.org",
+              "code": "57713-0",
+              "display": "INFANT FACTORS THAT AFFECT NEWBORN SCREENING INTERPRETATION"
+            }
+          ],
+          "text": "INFANT FACTORS THAT AFFECT NEWBORN SCREENING INTERPRETATION"
+        },
+        "subject": {
+          "reference": "Patient/09a7a7cc-7456-4f09-bdf6-ebc730902fd2",
+          "type": "Patient"
+        },
+        "effectiveDateTime": "2023-05-06T05:00:00-05:00",
+        "valueCodeableConcept": {
+          "coding": [
+            {
+              "system": "http://loinc.org",
+              "code": "LA12419-0",
+              "display": "INFANT IN NICU AT TIME OF SPECIMEN COLLECTION"
+            }
+          ],
+          "text": "INFANT IN NICU AT TIME OF SPECIMEN COLLECTION"
+        }
+      }
+    },
+    {
+      "fullUrl": "Observation/1c097524-defb-4dab-94c3-f603004ad7e3",
+      "resource": {
+        "resourceType": "Observation",
+        "id": "1c097524-defb-4dab-94c3-f603004ad7e3",
+        "status": "final",
+        "category": [
+          {
+            "coding": [
+              {
+                "system": "http://terminology.hl7.org/CodeSystem/observation-category",
+                "code": "social-history",
+                "display": "Social History"
+              }
+            ],
+            "text": "Social History"
+          }
+        ],
+        "code": {
+          "coding": [
+            {
+              "system": "https://www.health.state.mn.us",
```

I think a more realistic `system` for these local codes is `99MDH`.  That's what is in the ORM for the system.  It could be argued that ReportSteam could see `99MDH` and automatically map it to `https://www.health.state.mn.us` for us, but if there was to be a "pure" translation, it would be to `99MDH` directly.  Thoughts?

```suggestion
              "system": "99MDH",
```

Obviously, this comment applies to a few areas in this file.

---

Good call! That should be a URI.

---

Hrmm... the validator throws an error with that change:
> Bundle.entry[15].resource.ofType(Observation).code.coding[0]: Coding.system must be an absolute reference, not a local reference on line 487. [Jump to error.](http://localhost/validator/results#error-487)

---

Yeah, I'm not surprised that it threw an error since it isn't a URI.  I don't know how "bad" it is for there to be an error in the validator.   If we wanted to map it to the URI, we would have to reverse map it when we reconvert FHIR to HL7v2 OML.  The simplest thing that _might_ work is to set the `system` to `99MDH` such that it flows through the entire system without needing to do any mapping.

---

I'm thinking I'm OK with either way.  We'll just have to see how it all pans out with how RS can do the conversion and whether anything breaks once we do more intense levels of validation.